Automattic/delphin
# Framework: Add notice call to fail action in WPCOM Middleware

Follow up for #1000.

This PR tries to simplify `fail` action handling in WPCOM Middleware and always display error notice when fail action is executed. This would be a default behavior whenever you specify string or object as an action handler. See related discussion https://github.com/Automattic/delphin/pull/1000#discussion_r89943902.

It looks like we could simplify majority of `fail` action handlers with that change introduced. I can take care of it as soon as we agree that this change is something we all agree on.

### Testing
1. Make sure unit tests pass after this change.
2. Open https://delphin.live/?branch=update/extend-wpcom-middleware or http://delphin.localhost:1337.
3. Go to search page.
4. Disable your network connection by turning off wi-fi or if you use sandbox simply disconnect from it.
5. Click `Select` button.
6. You should see error notice informing about network connections.
7. Enable network connection again and make sure you are able to go to the next step.

### Review

/cc @Automattic/sdev-feed  @yurynix @scruffian @drewblaisdell
